pkg/geo/geomfn/subdivide.go

```diff
@@ -0,0 +1,309 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geomfn
+
+import (
+	"math"
+
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/cockroach/pkg/geo/geopb"
+	"github.com/cockroachdb/errors"
+	"github.com/twpayne/go-geom"
+)
+
+// Subdivide returns a geometry divided into parts, where each part contains no more than the number of vertices provided.
+func Subdivide(g geo.Geometry, maxVertices int) ([]geo.Geometry, error) {
+	if g.Empty() {
+		return []geo.Geometry{g}, nil
+	}
+	const minMaxVertices = 5
+	if maxVertices < minMaxVertices {
+		return []geo.Geometry{}, errors.Newf("max_vertices number cannot be less than %v", minMaxVertices)
```

Comment:
nit: return `nil` instead of empty array. same for below.

Reply:
Fixed